integration-tests/smoke/keeper_test.go

```diff
@@ -171,35 +173,82 @@ func getKeeperSuite(
 
 		Describe("with Keeper job", func() {
 			if testToRun == BasicSmokeTest {
-				It("performs upkeep of a target contract, stops upon cancel", func() {
-					// Let upkeep be performed atleast once
+				var testDescription = "check the initial upkeep (including the cancellation feature), then " +
+					"register another 9 upkeeps and watch all of them perform successfully"
+				It(testDescription, func() {
+					// Test that the upkeep which is registered in the BeforeEach function is executed
 					Eventually(func(g Gomega) {
-						cnt, err := consumer.Counter(context.Background())
-						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
-						g.Expect(cnt.Int64()).Should(BeNumerically(">", int64(0)), "Expected consumer counter to be greater than 0, but got %d", cnt.Int64())
-						log.Info().Int64("Upkeep counter", cnt.Int64()).Msg("Upkeeps performed")
+						oldestUpkeepCounter, err := consumer.Counter(context.Background())
+						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+						g.Expect(oldestUpkeepCounter.Int64()).Should(BeNumerically(">", int64(0)),
+							"Expected consumer counter to be greater than 0, but got %d", oldestUpkeepCounter.Int64())
+						log.Info().Int64("Upkeep counter", oldestUpkeepCounter.Int64()).Msg("Upkeeps performed")
 					}, "1m", "1s").Should(Succeed())
 
-					// Now cancel the upkeep as registry owner, it should get immediately cancelled
+					// Cancel the oldest upkeep via the registry
 					err := registry.CancelUpkeep(upkeepID)
 					Expect(err).ShouldNot(HaveOccurred(), "Upkeep should get cancelled successfully")
 					err = networks.Default.WaitForEvents()
 					Expect(err).ShouldNot(HaveOccurred(), "Error waiting for cancel upkeep tx")
 
-					// Get existing performed count
+					// Obtain the amount of times the upkeep has been executed so far
 					existingCnt, err := consumer.Counter(context.Background())
 					Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
 					log.Info().Int64("Upkeep counter", existingCnt.Int64()).Msg("Upkeep cancelled")
 
-					// Expect count to be remain consistent
+					// Expect the counter to remain constant because the upkeep was cancelled, so it shouldn't increase anymore
 					Consistently(func(g Gomega) {
 						cnt, err := consumer.Counter(context.Background())
 						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
 						g.Expect(cnt.Int64()).Should(
 							Equal(existingCnt.Int64()),
-							"Expected consumer counter to to remain constant at %d, but got %d", existingCnt.Int64(), cnt.Int64(),
+							"Expected consumer counter to remain constant at %d, but got %d", existingCnt.Int64(), cnt.Int64(),
 						)
 					}, "1m", "1s").Should(Succeed())
+
+					// Register another 9 upkeeps and deploy the contracts associated to them
+					registry, consumers, upkeepIDs := actions.DeployKeeperContracts(
```

Comment:
umm, this is gonna deploy a new registry. Can we create a function to just register an upkeep on existing registry?